specs/BrowserProcessExited.md

```diff
@@ -0,0 +1,186 @@
+unexpected process termination from the `ICoreWebView2Environment` interface so
+you can, e.g., cleanup the user data folder when it's no longer in use. In this
+document we describe the new API. We'd appreciate your feedback.
+
+
+# Description
+The `BrowserProcessExited` event allows developers to subscribe event handlers
+to be run when the `WebView2 Runtime`'s browser process associated to a
+`CoreWebView2Environment` terminates. A key scenario is cleanup of the use data
+folder used by the `WebView2 Runtime`, which is locked while the runtime's
+browser process is active.
+
+This event is raised for both expected and unexpected browser process
+termination. The `ICoreWebView2BrowserProcessExitedEventArgs` interfaces lets
+app developers get the `BrowserProcessExitKind` so they can decide how to handle
+different exit kinds or bypass handling if an event handler for the
+`CoreWebView2`s `ProcessFailed` event (for
+`CoreWebView2ProcessFailedKind.BrowserProcessFailed`) is already registered.
+
+
+# Examples
+The following code snippets demonstrate how the `BrowserProcessExited` event can
+be used:
+
+## Win32 C++
+```cpp
+// Before closing the WebView, register a handler with code to run once the
+// browser process is terminated.
+EventRegistrationToken browserExitedEventToken;
+
+CHECK_FAILURE(m_webViewEnvironment->add_BrowserProcessExited(
+    Callback<ICoreWebView2BrowserProcessExitedEventHandler>(
+        [browserExitedEventToken, this](
+            ICoreWebView2Environment* sender,
+            ICoreWebView2BrowserProcessExitedEventArgs* args) {
+        COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND kind;
+        CHECK_FAILURE(args->get_BrowserProcessExitKind(&kind));
+
+        // Watch for graceful browser process exit. Let ProcessFailed event
+        // handler take care of failed browser process termination.
+        if (kind == COREWEBVIEW2_BROWSER_PROCESS_EXIT_KIND_NORMAL_EXIT)
+        {
+          CHECK_FAILURE(
+              m_webViewEnvironment->remove_BrowserProcessExited(browserExitedEventToken));
+          // Release the environment only after the handler is invoked.
+          // Otherwise, there will be no environment to raise the event when
+          // the process exits.
+          m_webViewEnvironment = nullptr;
+          CleanupUserDataFolder();
+        }
+
+        return S_OK;
+    }).Get(),
+    &browserExitedEventToken));
+```
+
+## .NET C#
+```c#
+// Get the environment from the CoreWebView2 and add a handler.
+webView.CoreWebView2.Environment.BrowserProcessExited += Environment_BrowserProcessExited;
+
+// Check and report browser process exit kind.
+void Environment_BrowserProcessExited(object sender, CoreWebView2BrowserProcessExitedEventArgs e)
+{
+    var exitKind = (e.BrowserProcessExitKind == CoreWebView2BrowserProcessExitKind.NormalExit) ? "normally" : "unexpectedly";
+    MessageBox.Show(this, $"The browser process has exited {exitKind}.", "Browser Process Exited");
```

Comment:
We want to avoid sample code that just shows a message like this, in favor of what you did above in the COM code where it demonstrates an actual thing that an end developer wants to do. The goal with the sample code for the review docs is to validate that the API fits well with what an end developer will actually be doing with it, and accordingly we want sample code that shows that.

Reply:
I replaced this with a sample of runtime version update.